modules/core/src/hal_replacement.hpp

```diff
@@ -899,6 +899,27 @@ inline int hal_ni_meanStdDev(const uchar* src_data, size_t src_step, int width,
 #define cv_hal_meanStdDev hal_ni_meanStdDev
 //! @endcond
 
+/**
+ * @brief calculates dot product of two vectors (represented as 2d images)
+ *
+ * @param a_data Pointer to 1st 2nd image data
+ * @param a_step Stride of 1st 2nd image
+ * @param b_data Pointer to 1st 2nd image data
+ * @param b_step Stride of 1st 2nd image
+ * @param width Width of both images
+ * @param height Height of both images
+ * @param type Data type of both images, for example CV_8U or CV_32F
+ * @param dot_val Pointer to resulting dot product value
+ * @return int
+ */
+inline int hal_ni_dotProduct(const uchar* a_data, size_t a_step, const uchar* b_data, size_t b_step, int width, int height,
+                             int type, double *dot_val)
```

Comment:
can we return int64_t instead of double?

Reply:
Public interface: `double Mat::dot(InputArray _mat) const`. `int64_t` looks strange. Also the `::dot()` may handle floats and doubles.